pkg/odo/cli/component/create.go

```diff
@@ -477,6 +552,23 @@ func (co *CreateOptions) Validate() (err error) {
 
 // Run has the logic to perform the required actions as part of command
 func (co *CreateOptions) Run() (err error) {
+	if experimental.IsExperimentalModeEnabled() {
+		if co.devfileMetadata.devfileSupport {
```

Comment:
nit: can be a single if statement and reduce indent clutters

Reply:
Yeah, I know the logic can be optimized, the reason I leave all `experimental.IsExperimentalModeEnabled()` as independent condition is because it's better for readability (easy to separate experimental feature) and easy to remove since finally we don't need that condition.